pkg/scheduler/framework/plugins/dynamicresources/extendeddynamicresources.go

```diff
@@ -360,7 +413,17 @@ func createRequestsAndMappings(pod *v1.Pod, extendedResources map[v1.ResourceNam
 	// all mappings across all containers and resource types
 	var mappings []v1.ContainerExtendedResourceRequest
 
+	// Sort resource names to ensure deterministic ordering of device requests and mappings.
+	// Maps have non-deterministic iteration order in Go, so we extract and sort the keys.
+	resourceNames := make([]v1.ResourceName, 0, len(extendedResources))
 	for resource := range extendedResources {
+		resourceNames = append(resourceNames, resource)
+	}
+	sort.Slice(resourceNames, func(i, j int) bool {
+		return resourceNames[i] < resourceNames[j]
+	})
```

Comment:
why is this needed? each resource is independent of each other, this sorting does not help.